ts4.1/icu.d.ts

```diff
@@ -0,0 +1,85 @@
+declare module 'react-i18next/icu.macro' {
+  type Namespace = import('.').Namespace;
+  type DefaultNamespace = import('.').DefaultNamespace;
+  type Resources = import('.').Resources;
+  type TFuncKey<N extends Namespace = DefaultNamespace, T = Resources> = import('.').TFuncKey<N, T>;
+  type i18n = import('i18next').i18n;
```

Comment:
Everything inside the module will be exported (even without the `export` keyword), I don't think we have an `i18n` under `react-i18next`. If you don't want to re-export all these types, you should import them outside the module.
```suggestion
import {Namespace, DefaultNamespace, Resources, TFuncKey} from '.';
import {i18n} from 'i18next';

declare module 'react-i18next/icu.macro' {
```

Reply:
when I did it this way, the types stopped working in VS Code, and it also failed to type them at all (they became `any`). Can you confirm?

Reply:
yeah, it works perfectly, but you also need to rename the file declaration to `uci.macro.d.ts`, otherwise, it won't find the module.

<img width="680" alt="Screenshot 2021-05-08 at 1 14 01 PM" src="https://user-images.githubusercontent.com/12190482/117549310-5014e500-afff-11eb-8d54-8340a329cff6.png">

Reply:
FYI, tslint is going nuts now. Should we be concerned? I'm going to push this as you recommend, and then we can decide this question.

<img width="844" alt="Screen Shot 2021-05-08 at 9 09 15 PM" src="https://user-images.githubusercontent.com/98250/117557534-a73ba980-b041-11eb-9706-adf590debe24.png">

Reply:
So you can add the export again if you want